lib/src/core/message/guild_emoji.dart

```diff
@@ -1,40 +1,66 @@
-import 'package:nyxx/src/nyxx.dart';
-import 'package:nyxx/src/core/snowflake.dart';
-import 'package:nyxx/src/core/snowflake_entity.dart';
-import 'package:nyxx/src/core/guild/guild.dart';
-import 'package:nyxx/src/core/guild/role.dart';
-import 'package:nyxx/src/core/message/emoji.dart';
+import 'package:nyxx/nyxx.dart';
 import 'package:nyxx/src/internal/cache/cacheable.dart';
-import 'package:nyxx/src/typedefs.dart';
 
 abstract class IBaseGuildEmoji implements SnowflakeEntity, IEmoji {
   /// True if emoji is partial.
   bool get isPartial;
 
-  /// Returns cdn url to emoji
-  String get cdnUrl;
+  /// The name of the emoji.
+  String get name;
+
+  /// Whether this emoji is animated.
+  bool get animated;
 
   /// Creates partial emoji from given String or Snowflake.
-  factory IBaseGuildEmoji.fromId(Snowflake id) => GuildEmojiPartial(id);
+  factory IBaseGuildEmoji.fromId(Snowflake id) => GuildEmojiPartial({"id": id});
+
+  /// Returns cdn url to emoji
+  String cdnUrl({String? format = "webp", int? size = 128});
 }
 
 abstract class BaseGuildEmoji extends SnowflakeEntity implements IBaseGuildEmoji {
   /// True if emoji is partial.
   @override
   bool get isPartial;
 
-  /// Returns cdn url to emoji
+  /// Whether this emoji is animated.
   @override
-  String get cdnUrl => "https://cdn.discordapp.com/emojis/$id.png";
+  bool get animated;
+
+  /// The name of the emoji.
+  @override
+  String get name;
 
   /// Creates an instance of [BaseGuildEmoji]
   BaseGuildEmoji(RawApiMap raw) : super(Snowflake(raw["id"]));
 
+  /// Returns cdn url to emoji
+  @override
+  String cdnUrl({String? format = "webp", int? size = 128}) {
```

Comment:
```suggestion
  String cdnUrl({String? format, int? size}) {
```
It needs tp be null by deafault because otherwise it wont do thing automatically